src/components/Task/Task.jsx

```diff
@@ -0,0 +1,17 @@
+import { Link } from 'react-router-dom';
+import './Task.scss';
+
+export const Task = ({id, title, status, changeStatus, userData}) => {
+    const data = {
+        userId: userData.uid,
+        taskId: id
+    }
+    return (
+        <div className={`task ${status && 'task_done'}`}>
+            <button className={`task__button ${status && 'task__button_done'}`} onClick={() => changeStatus(data)}></button>
```

Comment:
Функцию вынести вне return

const handleChangeStatus = () => changeStatus(data)

```suggestion
            <button className={`task__button ${status && 'task__button_done'}`} onClick={handleChangeStatus}></button>
```